src/System.Security.Cryptography.Primitives/tests/HashAlgorithmNameTests.cs

```diff
@@ -0,0 +1,59 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+using Xunit;
+
+namespace System.Security.Cryptography.Primitives.Tests
+{
+    public static class HashAlgorithmNameTests
+    {
+        [Theory]
+        [MemberData(nameof(ValidInputs))]
+        public static void FromOid_ValidInput(string oid, HashAlgorithmName expected)
+        {
+            Assert.Equal(expected, HashAlgorithmName.FromOid(oid));
+        }
+
+        [Fact]
+        public static void TryFromOid_ThrowsForNullInput()
+        {
+            Assert.Throws<ArgumentNullException>(() => HashAlgorithmName.FromOid(null));
+        }
+
+        [Fact]
+        public static void TryFromOid_ThrowsForInvalidInput()
+        {
+            Assert.Throws<CryptographicException>(() => HashAlgorithmName.FromOid("1.2.3.4"));
+        }
+
+        [Theory]
+        [MemberData(nameof(ValidInputs))]
+        public static void TryFromOid_ValidInput(string oid, HashAlgorithmName expected)
+        {
+            Assert.True(HashAlgorithmName.TryFromOid(oid, out HashAlgorithmName actual));
+            Assert.Equal(expected, actual);
+        }
+
+        [Theory]
+        [InlineData(null)]
+        [InlineData("1.2.3.4")]
```

Comment:
"SHA1" (no friendly name coercion), "1.2.840.113549.1.1.1" (known OID (rsaEncryption), not a hash algorithm).

Reply:
I went with sha1WithRsa OID to make it clear with tests that I didn't intend for this to handle those, even though they are "digest-y".